lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -437,6 +438,69 @@ def delete(self, trans, id, **kwd):
         # TODO: Unsure of response message to let api know that a workflow was successfully deleted
         return ("Workflow '%s' successfully deleted" % stored_workflow.name)
 
+    @expose_api
+    def sharing(self, trans, id, **kwargs):
+        """
+        GET /api/workflows/sharing
```

Comment:
Clearly this method is doing more than `Return a workflow with the given id.`, so I'd like to see those things properly documented
I'm not a fan of copy-pasting controller stuff, which make very few guarantees about being stable, into the API, which should be there permanently across releases. To me all those predicates in kwargs should ideally be separate routes, and for instance returning the wokflow list when requesting `unshare_me` is suboptimal.

Reply:
I have just started reviewing this, more changes will come. Thanks for the ideas!